### PR TITLE
codegen: only use ret if a builder_postprocess is set

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
this allows us to enable clippy::let_and_return lint

### CODE REVIEW

**Comment by sdroege** [comment]
Uh....

```
 Error:   --> gio/src/auto/converter_output_stream.rs:76:48
   |
76 | pub const NONE_CONVERTER_OUTPUT_STREAM: Option<&ConverterOutputStream> = None;
   |                                                ^ expected named lifetime parameter
   |
help: consider using the `'static` lifetime
   |
76 | pub const NONE_CONVERTER_OUTPUT_STREAM: Option<&'static ConverterOutputStream> = None;
   |                                                 ^^^^^^^
help: consider introducing a named lifetime parameter
   |
40 | impl<'a> ConverterOutputStreamBuilder {
41 |     pub fn new() -> Self {
42 |         Self::default()
43 |     }
44 | 
45 | 
 ...
```

Where does that suddenly come from? Also why did this compile before? :)

**Review by sdroege — APPROVED** [review]

File: src/codegen/object.rs [hunk 1]
```diff
@@ -281,16 +281,26 @@ impl {}Builder {{
...
+    } else {
+        writeln!(
+            w,
+            r#"        {}::Object::new::<{}>(&properties)
+                .expect("Failed to create an instance of {}")"#,
+            glib_crate_name, analysis.name, analysis.name,
+        )?;
     }
-    writeln!(w, "    ret\n    }}")?;
+
```
Comment (sdroege) [inline src/codegen/object.rs:303]:
```
/usr/share/rust/.cargo/bin/cargo fmt -- --check
Diff in /home/runner/work/gir/gir/src/codegen/object.rs at line 281:
     } else {
         "glib"
     };
-    
+
     // The split allows us to not have clippy::let_and_return lint disabled
     if let Some(code) = analysis.builder_postprocess.as_ref() {
         writeln!(
```

**Comment by bilelmoussaoui** [comment]
> Uh....
> 
> ```
>  Error:   --> gio/src/auto/converter_output_stream.rs:76:48
>    |
> 76 | pub const NONE_CONVERTER_OUTPUT_STREAM: Option<&ConverterOutputStream> = None;
>    |                                                ^ expected named lifetime parameter
>    |
> help: consider using the `'static` lifetime
>    |
> 76 | pub const NONE_CONVERTER_OUTPUT_STREAM: Option<&'static ConverterOutputStream> = None;
>    |                                                 ^^^^^^^
> help: consider introducing a named lifetime parameter
>    |
> 40 | impl<'a> ConverterOutputStreamBuilder {
> 41 |     pub fn new() -> Self {
> 42 |         Self::default()
> 43 |     }
> 44 | 
> 45 | 
>  ...
> ```
> 
> Where does that suddenly come from? Also why did this compile before? :)

isn't that just another issue that will disappear after re-running the CI because dark magic?

**Comment by sdroege** [comment]
> isn't that just another issue that will disappear after re-running the CI because dark magic?

It's unrelated to this PR but apparently it happened again...

**Pull request merged**